oneandone/src/main/java/org/apache/jclouds/oneandone/rest/config/OneAndOneHttpApiModule.java

```diff
@@ -0,0 +1,116 @@
+import org.jclouds.rest.ConfiguresHttpApi;
+import org.jclouds.rest.config.HttpApiModule;
+import static org.jclouds.util.Predicates2.retry;
+
+@ConfiguresHttpApi
+public class OneAndOneHttpApiModule extends HttpApiModule<OneAndOneApi> {
+
+    @Override
+    protected void bindErrorHandlers() {
+        bind(HttpErrorHandler.class).annotatedWith(Redirection.class).to(OneAndOneHttpErrorHandler.class);
+        bind(HttpErrorHandler.class).annotatedWith(ClientError.class).to(OneAndOneHttpErrorHandler.class);
+        bind(HttpErrorHandler.class).annotatedWith(ServerError.class).to(OneAndOneHttpErrorHandler.class);
+    }
+
+    @Override
+    protected void configure() {
+        super.configure();
+        bind(DateAdapter.class).to(Iso8601DateAdapter.class);
+    }
+
+    @Provides
+    @Named(POLL_PREDICATE_PRIVATE_NETWORK)
+    Predicate<ServerPrivateNetworkRef> providePrivateNetworkReadyPredicate(final OneAndOneApi api, OneAndOneConstants constants) {
+        return retry(new PrivateNetworkReadyPredicate(
+                api),
+                constants.pollTimeout(), constants.pollPeriod(), constants.pollMaxPeriod(), TimeUnit.SECONDS);
+    }
+
+    @Provides
+    @Named(POLL_PREDICATE_SERVER)
+    Predicate<Server> provideServerReadyPredicate(final OneAndOneApi api, OneAndOneConstants constants) {
+        return retry(new ServerReadyPredicate(
+                api),
+                constants.pollTimeout(), constants.pollPeriod(), constants.pollMaxPeriod(), TimeUnit.SECONDS);
+    }
+
+    static class ServerReadyPredicate implements Predicate<Server> {
+
+        private final OneAndOneApi api;
+
+        public ServerReadyPredicate(OneAndOneApi api) {
+            this.api = checkNotNull(api, "api must not be null");
+        }
+
+        @Override
+        public boolean apply(Server server) {
+            checkNotNull(server, "Server");
+            server = api.serverApi().get(server.id());
+            if ((server.status().state() != Types.ServerState.POWERED_OFF
+                    && server.status().state() != Types.ServerState.POWERED_ON)
+                    || server.status().percent() != 0) {
+                return false;
+            } else {
+                return true;
+            }
+        }
+    }
+
+    static class PrivateNetworkReadyPredicate implements Predicate<ServerPrivateNetworkRef> {
+
+        private final OneAndOneApi api;
+
+        public PrivateNetworkReadyPredicate(OneAndOneApi api) {
+            this.api = checkNotNull(api, "api must not be null");
+        }
+
+        @Override
+        public boolean apply(ServerPrivateNetworkRef networkRef) {
+            checkNotNull(networkRef, "ServerPrivateNetworkRef");
+            PrivateNetwork server = api.serverApi().getPrivateNetwork(networkRef.serverId(), networkRef.privateNetworkId());
+
+            if (server.state() != Types.GenericState.ACTIVE) {
+                return false;
+            } else {
+                return true;
+            }
```

Comment:
Just `return server.state() == Types.GenericState.ACTIVE`? Apply this to the other similar conditionals.